addons/website_slides_survey/tests/test_course_certification_stats.py

```diff
@@ -0,0 +1,98 @@
+        super().setUpClass()
+
+        # Create certifications
+        cls.certifications = cls.env["survey.survey"].create(
+            [
+                {
+                    "title": f"certification {i + 1}",
+                    "scoring_type": "scoring_with_answers",
+                    "certification": True,
+                }
+                for i in range(3)
+            ]
+        )
+
+        # Create courses and link them to certifications
+        [cls.course_1, cls.course_2, cls.course_3] = cls.env["slide.channel"].create(
+            [
+                {
+                    "name": f"Course {i +1}",
+                    "enroll": "public",
+                    "is_published": True,
+                    "slide_ids": [Command.create(
+                        {
+                            "name": f"Certification {i + 1} slide",
+                            "slide_category": "certification",
+                            "survey_id": cls.certifications.ids[i],
+                            "is_published": True,
+                        })]
+                }
+                for i in range(3)
+            ]
+        )
+
+        # Create course participants
+        cls.participants = cls.env["res.users"].browse([
+            cls.survey_user.id,
+            cls.user_emp.id,
+            cls.user_portal.id
+        ])
+        cls.courses = cls.env["slide.channel"].browse([
+            cls.course_1.id,
+            cls.course_2.id,
+            cls.course_3.id
+        ])
+        cls.partner_memberships = cls.courses._action_add_members(cls.participants.partner_id)
+
+        # Set certified/not certified participants
+        cls.slide_partners = cls.env["slide.slide.partner"].create(
+            [
+                {
+                    "channel_id": slide.channel_id.id,
+                    "partner_id": partner.id,
+                    "slide_id": slide.id,
+                }
+                for slide in cls.courses.slide_ids
+                for partner in cls.participants.partner_id
+            ]
+        )
+
+        cls.slide_partners[0].survey_scoring_success = True  # survey_user certified for course_1
+        cls.slide_partners[-1].survey_scoring_success = True  # user_portal certified for course_3
+        cls.slide_partners._recompute_completion()  # update slide_channel_partner.survey_certification_success
+
+    def test_channel_members_certified_count(self):
+        self.assertEqual(
+            self.course_1.members_certified_count,
+            1,
+            "Course 1 has 1 certified participant",
+        )
+        self.assertEqual(
+            self.course_2.members_certified_count,
+            0,
+            "Course 2 has 0 certified participant",
+        )
+        self.assertEqual(
+            self.course_3.members_certified_count,
+            1,
+            "Course 3 has 1 certified participant",
+        )
+
+    def test_channel_partner_survey_certification_success(self):
```

Comment:
I think we can regroup both testing methods as they're short and call the method "test_membership_certification_fields" or something like that.